vault/resource_kubernetes_secret_backend.go

```diff
@@ -0,0 +1,179 @@
+package vault
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+	"github.com/hashicorp/terraform-provider-vault/util"
+)
+
+const (
+	fieldKubernetesHost    = "kubernetes_host"
+	fieldKubernetesCACert  = "kubernetes_ca_cert"
+	fieldServiceAccountJWT = "service_account_jwt"
+	fieldDisableLocalCAJWT = "disable_local_ca_jwt"
+)
+
+func kubernetesSecretBackendResource() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: kubernetesSecretBackendCreateUpdate,
+		ReadContext:   kubernetesSecretBackendRead,
+		UpdateContext: kubernetesSecretBackendCreateUpdate,
+		DeleteContext: kubernetesSecretBackendDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: getKubernetesSecretsMountSchema(),
+	}
+}
+
+func getKubernetesSecretsMountSchema() map[string]*schema.Schema {
+	s := getMountSchema("type")
+	for k, v := range getKubernetesSecretsSchema() {
+		s[k] = v
+	}
+	return s
+}
+
+func getKubernetesSecretsSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		fieldKubernetesHost: {
+			Type:        schema.TypeString,
+			Description: "The Kubernetes API URL to connect to.",
+			Optional:    true,
+			Computed:    true,
+		},
+		fieldKubernetesCACert: {
+			Type: schema.TypeString,
+			Description: "A PEM-encoded CA certificate used by the secret engine to " +
+				"verify the Kubernetes API server certificate. Defaults to the " +
+				"local pod’s CA if found, or otherwise the host's root CA set.",
+			Optional: true,
+			Computed: true,
```

Comment:
Will Vault return the pod's CA if we no CA is specified?

Reply:
Ah, no it will not. This probably shouldn't be computed then, right?